specs/Workers.md

```diff
@@ -0,0 +1,1228 @@
+
+**What is Web Worker?**
+[Web Workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API) allow web content to execute scripts in background threads, enabling tasks to run independently of the user interface and facilitating communication between threads using message passing. Web Workers can also make network requests using fetch() or XMLHttpRequest APIs. Few examples of workers are
+
+- **Dedicated Worker**: Dedicated Workers are a type of Web Worker that operates independently in the background of a web page, enabling parallel execution of tasks without blocking the main UI thread. They are created by specific web pages and communicate with them via message passing, making them ideal for offloading CPU-intensive operations and improving overall performance and responsiveness of web applications.
+
+- **Shared Worker**: Shared Workers are a type of Web Worker that can be accessed by multiple scripts running in different windows, IFrames, or other contexts, as long as they are within the same domain as the worker. Unlike Dedicated Workers, which are tied to a specific web page, Shared Workers require communication via an active port, making them slightly more complex to implement but enabling collaboration and shared resources across different parts of a web application.
+
+- **Service Worker**: Service Workers serve as intermediary agents between web applications, the browser, and the network, functioning as proxy servers. Primarily, they facilitate the development of seamless offline experiences by caching resources and enabling web apps to continue functioning without an internet connection. Additionally, Service Workers intercept network requests, enabling customized responses based on network availability, and can update server-side assets. Moreover, they grant access to push notifications and background sync APIs, enhancing the functionality and interactivity of web applications.
+
+# Description
+
+We propose the introduction of APIs that enable the host application to oversee workers. These APIs lay the groundwork for the development of additional APIs, facilitating direct interaction between the host application and the workers, and support background operations for service workers.
+
+**DedicatedWorkerCreated**: This event, associated with CoreWebView2, is triggered when a web page initiates a dedicated worker. It grants access to the CoreWebView2DedicatedWorker object, which contains information such as the script URL and name. Additionally, it provides a 'destroy' event that is fired just before this object is due for destruction.
```

Comment:
We say events are 'raised' rather than triggered or fired or other in Microsoft docs. Applies below as well.

```suggestion
**DedicatedWorkerCreated**: This event, associated with CoreWebView2, is raised when a web page initiates a dedicated worker. It grants access to the CoreWebView2DedicatedWorker object, which contains information such as the script URL and name. Additionally, it provides a 'destroy' event that is fired just before this object is due for destruction.
```